sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp

```diff
@@ -68,7 +68,9 @@ convertMatrixUseStringToEnum(const char *UseString) {
   return std::nullopt;
 }
 
-inline __SYCL_ALWAYS_INLINE __spv::MatrixLayout joint_matrix_layout_to_spv(
+// propagateConstexprLayout uses the exact name of the function, so we use
+// extern "C" here.
+extern "C" constexpr __spv::MatrixLayout joint_matrix_layout_to_spv(
```

Comment:
Do we not need the inlining attribute anymore?

Reply:
No, as `constexpr` also includes `inline`, see [the comment from Alexey](https://github.com/intel/llvm/pull/16628#discussion_r1915080267)

Reply:
Ah...good...sorry..missed that as it was hiding.